satpy/writers/cf_writer.py

```diff
@@ -499,14 +514,20 @@ def da2cf(dataarray, epoch=EPOCH, flatten_attrs=False, exclude_attrs=None, compr
                 If True, flatten dict-type attributes
             exclude_attrs (list):
                 List of dataset attributes to be excluded
-
+            valid_cf_dataset_name (bool):
+                Prepend dataset name with CHANNEL_ if staring with a digit
         """
         if exclude_attrs is None:
             exclude_attrs = []
 
         new_data = dataarray.copy()
         if 'name' in new_data.attrs:
             name = new_data.attrs.pop('name')
+            if valid_cf_dataset_name and name[0].isdigit():
+                _orig_name = name
+                name = 'CHANNEL_' + name
+                warnings.warn('Rename dataset {} to {}.'.format(_orig_name, name))
+                new_data.attrs['satpy_dataset_name'] = _orig_name
```

Comment:
Does this attribute show up in the resulting NetCDF file? Does @mraspaud recent work with wavelength range and all that change how this could/should work?

Reply:
Yes this show up in the netcdf file.

I would need to dig into this investigating if this can be skipped.

Reply:
So, I did not understand that with the wavelength work by @mraspaud you actually don't need the dataset or variable name.

What happened, I messed up my code. I thought I added renaming of the dataset back to the original when reading back the netcdf file. But I messed up my git branch, and removed by accident this when resetting to a previous commit. So I thought my code worked, but it was the work of @mraspaud fixing this. If this make any sense. Anyway this makes it a lot simpler I think. Will try to clean up this tomorrow.